src/components/AvatarGroup/index.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable react/no-array-index-key */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Avatar } from '../';
+import StyledMoreAvatar from './styled/moreAvatar';
+import StyledContainer from './styled/container';
+
+/**
+ * Avatar groups are used to bunch together avatars
+ */
+export default function AvatarGroup(props) {
+    const { size, className, style, avatars, maxAvatars, showCounter } = props;
+
+    const spacing = '-6px';
+
+    const renderMoreItems = (max, total) => {
+        if (total <= max || !showCounter) return null;
+
+        const moreItems = avatars.slice(max).length;
+
+        return (
+            <StyledMoreAvatar style={{ zIndex: 0, marginLeft: `${spacing}` }}>
+                <span>{`+${moreItems}`}</span>
+            </StyledMoreAvatar>
+        );
+    };
+
+    const total = avatars.length;
+    const maxAvatar = total > maxAvatars && showCounter ? maxAvatars - 1 : maxAvatars;
+
+    const items = avatars
+        .slice(0, maxAvatar)
+        .map((avatar, idx) => (
+            <Avatar
+                {...avatar.props}
+                size={size}
+                style={{ zIndex: `${maxAvatar - idx}`, marginLeft: `${spacing}` }}
+                key={idx}
+            />
+        ));
+
+    return (
+        <StyledContainer className={className} style={style} role="group">
+            {items}
+            {renderMoreItems(+maxAvatar, total)}
```

Comment:
Instead of make iteration in this component let's break this into more components like:
```
return (
    <StyledContainer className={className} style={style}>
        <RenderIf isTrue={shouldRenderCounter}>
             <Counter />
        </RenderIf>
        <Avatars maxAvatar={maxAvatar} total={total} />
    </StyledContainer>
);
```